src/tests/possumwood/subnetwork_from_file.cpp

```diff
@@ -0,0 +1,144 @@
+#include <boost/test/unit_test.hpp>
+
+#include <dependency_graph/graph.h>
+#include <dependency_graph/metadata_register.h>
+#include <dependency_graph/rtti.h>
+#include <dependency_graph/node_base.inl>
+#include <dependency_graph/nodes.inl>
+#include <dependency_graph/port.inl>
+
+#include <actions/actions.h>
+
+#include <actions/filesystem_mock.h>
+#include <possumwood_sdk/app.h>
+
+#include "common.h"
+
+using namespace dependency_graph;
+using possumwood::io::json;
+
+namespace {
+
+dependency_graph::NodeBase& findNode(dependency_graph::Network& net, const std::string& name) {
+	for(auto& n : net.nodes())
+		if(n.name() == name)
+			return n;
+
+	BOOST_REQUIRE(false && "Node not found, fail");
+	throw;
+}
+
+dependency_graph::NodeBase& findNode(const std::string& name) {
+	return findNode(possumwood::AppCore::instance().graph(), name);
+}
+
+json readJson(possumwood::IFilesystem& filesystem, const std::string& filename) {
+	json result;
+
+	auto stream = filesystem.read(possumwood::Filepath::fromString(filename));
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/martin.prazak/possumwood/pullRequest?prid=6093062)